### PR TITLE
Improve how `sf` is bundled with the `sfdx` installers

### PR DESCRIPTION
**Pull request by RodEsp**
### What does this PR do?
* Makes `sf` available to the user without any `npm` commands when installing `sfdx` with the installers. 
    * Pulls in the new version of `oclif` that allows for the creation of Mac and Windows installers that make more than one Oclif CLI available on the user's machine.
* Changes the `postupdate` hook for `sfdx update` to no longer install `sf` using `npm`. It now just uses the version of `sf` that comes pre-bundled with the tarballs/installers.

### Testing
See https://github.com/oclif/oclif/pull/696 for testing instructions

### What issues does this PR fix or reference?
@W-9845364@

#### Depends on https://github.com/oclif/oclif/pull/696 and https://github.com/salesforcecli/plugin-release-management/pull/353.

### CODE REVIEW

File: package.json
Comment (RodEsp) [inline package.json:181]: This will be updated to the correct version once the oclif PR is merged.

File: package.json
Comment (RodEsp) [inline package.json:151]: This will be updated to the correct version once the `plugin-release-management` PR is merged.

File: .circleci/config.yml
Comment (mshanemc) [inline .circleci/config.yml:352]: `release-package` requires `approval` so adding it here (and below) is redundant, no?

File: package.json
Comment (mshanemc) [reply package.json:151]: that merged, please update

File: package.json
Comment (mshanemc) [inline package.json:202]: please explain this one to me.

**Review by mshanemc — APPROVED** [review]
non-blocking comments.
still waiting for updates from oclif/relMgmt versions

File: package.json
Comment (peternhale) [reply package.json:202]: @mshanemc I believe the command will hang on windows if the dot is not present.

Comment (RodEsp) [reply package.json:202]: This is all the explanation I got: https://salesforce-internal.slack.com/archives/G02K6C90RBJ/p1635790670258600?thread_ts=1635183450.084400&cid=G02K6C90RBJ

Builds were hanging on that step without it, working with it. But not on Windows, on the CircleCI Linux machines.

File: .circleci/config.yml
Comment (RodEsp) [reply .circleci/config.yml]: Just a little extra safe-guarding in case anyone removes `release-package` as a required job here for whatever purpose (testing, dev deployments, etc...)

Comment (RodEsp) [reply .circleci/config.yml]: Oh, nvm, Pete removed these.

**Pull request merged**
